aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -85,17 +85,26 @@ impl AptosVMImpl {
             0 => None,
             _ => StorageGasSchedule::fetch_config(&storage),
         };
-        if gas_feature_version >= 2 {
-            if let (Some(gas_params), Some(storage_gas_schedule)) =
-                (&mut gas_params, &storage_gas_schedule)
-            {
+
+        if let (Some(gas_params), Some(storage_gas_schedule)) =
+            (&mut gas_params, &storage_gas_schedule)
+        {
+            if gas_feature_version >= 2 {
                 gas_params.natives.table.common.load_base =
                     storage_gas_schedule.per_item_read.into();
                 gas_params.natives.table.common.load_per_byte =
                     storage_gas_schedule.per_byte_read.into();
                 gas_params.natives.table.common.load_failure = 0.into();
             }
+
+            if gas_feature_version >= 7 {
+                gas_params.natives.aptos_framework.object.exists_at.per_item =
```

Comment:
Why do we want to use these values from the storage schedule?

Reply:
To make them consistent with the way resource loading is handled for the global operations (exists, borrow_global etc.)

The reason why we have to duplicate these parameters for some native functions is due to the VM's lack of a unified memory manager. The VM doesn't get notified when a native function accesses a global value, so such native would have to do the bookkeeping itself. This is something I'm planning to solve with a new architecture, but it will take some time to implement that.